docs/adr/0024-upgrade-templating-toolset-of-auth-server-emails.md

```diff
@@ -0,0 +1,261 @@
+  - React offers us many integration options. E.g there is one with [MJML](https://github.com/wix-incubator/mjml-react)
+  - React has already been audited by our security team.
+- Cons
+  - It could be a overkill to use React, since the library comes with far too many features which may not be needed for templating
+  - The templates in the `fxa-auth-server` do not contain heavy business logic so building them with a frontend-heavy framework like React might complicate things
+  - Since we will be converting React components into static HTML pages, another layer of complexity as well as new dependencies would be added to the project
+
+### Option C - Use EJS
+
+- Description
+  - Template the auth server mail components using [EJS](https://ejs.co/)
+- Pros
+  - The syntax and logic of EJS is similar to JavaScript, leading to a very small learning curve.
+  - EJS has a large community of users which will ensure faster help and support.
+  - It is a simple templating engine and far easier to use as its plain Javascript and since we want to keep our mail templates straightforward, using EJS could be a good choice.
+- Cons
+  - Adds a new dependency to the stack
+  - Lacks support for `block` functionality, which will make resuing components a bit difficult.
+  - Although we can implement `block` functionality with the help of third-party support, most of the projects are outdated and not maintained anymore.
+
+### Pros and Cons of using MJML as part of our templating stack
+
+As part of modernizing the templates, we aim to replace the framework in which our templates are currently built in so as to include more logic in them while development. Along with the framework, we considered using [MJML](https://mjml.io/) as our potential markup language, designed to reduce the pain of coding a responsive email.
+
+- Pros
+  - MJML is a component-based library having its own pre-built components as well as gives us the ability to define custom compononents which translates to responsive HTML after compilation. These components can be reused across the templates and modified by passing parameters e.g. [this social links component](https://documentation.mjml.io/#mj-social)
+  - MJML can help us create responsive emails using mobile-first approach and it controls that responsiveness automatically needing very less to no developer intervention.
+  - This library has been created by keeping in mind all the industry standards as well as the quirks of old email clients, so we will eventually be saving time on investigating cross-browser compatability of our templates
+  - MJML has by far the highest weekly downloads and github stars in comparision to its alternatives like [Foundation for Emails](https://github.com/foundation/foundation-emails).
+  - MJML can be integrated with React to generate responsive emails and there is also a [library](https://github.com/wix-incubator/mjml-react) that could help us with that purpose.
```

Comment:
Might be worth adding that there's MJML previewers as well, so it can be used for previewing as well.

Reply:
Just for cross-reference, I asked about the previewing options here: https://github.com/mozilla/fxa/pull/8765#discussion_r646991714

But I _think_ you answered that here: https://github.com/mozilla/fxa/pull/8765#issuecomment-856312379